docs/reference/hoconOverview.md

```diff
@@ -3,82 +3,158 @@ id: hoconOverview
 
+Data pipelines are defined with four sections:
 
-## Overview
-The configuration files are stored in the [HOCON](https://github.com/lightbend/config/blob/master/HOCON.md) file format.
-The main sections are global, connections, data objects and actions.
-As a starting point, use the [application.conf](https://github.com/smart-data-lake/sdl-examples/blob/develop/src/main/resources/application.conf) from SDL-examples.
-More details and options are described below.
+* [**global options**](#global-options) variables e.g. Spark settings
+* [**connections**](#connections): defining the settings for external sources or targets
+* [**data objects**](#data-objects): including type, format, location etc., and 
+* [**actions**](#actions), which describes how to get from one to another dataObject, including transformations
+
+The settings are structured in a hierarchy. 
 
-### Global Options
-In the global section of the configuration you can set global configurations such as spark options used by all executions.
-You can find a list of all configurations in the [Configuration Schema Browser](https://smartdatalake.ch/json-schema-viewer/index.html#viewer-page?v=0).  
-A good list with examples can be found in the [application.conf](https://github.com/smart-data-lake/sdl-examples/blob/develop/src/main/resources/application.conf) of sdl-examples.
+The [**Configuration Schema Viewer**](../../JsonSchemaViewer) presents all available options in each category with all the available parameters for each element. 
 
-### Secrets User and Password Variables
-TODO:
-Mention Azure Key Vault here or at least point to the Azure specific doc to be created.
+Furthermore, there are the following general (Hocon) features:
 
-Usernames, passwords and other secrets should not be stored in your configuration files in clear text as these files are often stored directly in the version control system.
-They should also not be visible in logfiles after execution.
+* [handling of secrets](hoconSecrets)
+* [handling of variables](hoconVariables)
 
-Instead of having the username and password configured directly in your configuration files, you can use secret providers
-for configuration values that end with `...Variable`, like BasicAuthMode's userVariable and passwordVariable.
-To configure a secret use the convention `&ltSecretProviderId&gt#&ltSecretName&gt`.
+The pipeline definition can be separated in multiple files and directories, specified with the SDLB option `-c, --config <file1>[,<file2>...]`. This could also be a list of directories or a mixture of directories and files. All configuration files (`*.conf`) within the specified directories and its subdirectories are taken into account. 
 
-Default secret providers are:
+> Note: Also `.json` is an excepted file extension. Files with other extensions are disregarded. 
```

Comment:
Isn't .properties accepted as well?

Reply:
correct. I tried successfully and added the note.